src/AltinnCore/Runtime/RequestHandling/MultipartRequestReader.cs

```diff
@@ -0,0 +1,131 @@
+        /// Gets a list of all parts.
+        /// </summary>
+        public List<RequestPart> Parts { get; }
+
+        /// <summary>
+        /// Gets a list of errors.
+        /// </summary>
+        public List<string> Errors { get; }
+
+        /// <summary>
+        /// Process the request and generate parts.
+        /// </summary>
+        /// <returns>A <see cref="Task"/> representing the result of the asynchronous operation.</returns>
+        public async Task Read()
+        {
+            if (IsMultipart)
+            {
+                int partCounter = 0;
+                try
+                {
+                    MultipartReader reader = new MultipartReader(GetBoundary(), request.Body);
+
+                    MultipartSection section;
+                    while ((section = await reader.ReadNextSectionAsync()) != null)
+                    {
+                        partCounter++;
+
+                        bool hasContentDispositionHeader = ContentDispositionHeaderValue
+                               .TryParse(section.ContentDisposition, out ContentDispositionHeaderValue contentDisposition);
+
+                        if (!hasContentDispositionHeader)
+                        {
+                            Errors.Add(string.Format("Part number {0} doesn't have a content disposition", partCounter));
+                            continue;
+                        }
+
+                        if (section.ContentType == null)
+                        {
+                            Errors.Add(string.Format("Part number {0} doesn't have a content type", partCounter));
+                            continue;
+                        }
+
+                        string sectionName = contentDisposition.Name.HasValue ? contentDisposition.Name.Value : null;
+                        string contentFileName = contentDisposition.FileName.HasValue ? contentDisposition.FileName.Value : null;
+                        long fileSize = contentDisposition.Size ?? 0;
+
+                        MemoryStream memoryStream = new MemoryStream();
+                        section.Body.CopyTo(memoryStream);
+                        memoryStream.Position = 0;
+
+                        Parts.Add(new RequestPart()
+                        {
+                            ContentType = section.ContentType,
+                            Name = sectionName,
+                            Stream = memoryStream,
+                            FileName = contentFileName,
+                            FileSize = fileSize,
+                        });
+                    }
+                }
+                catch (IOException ioex)
+                {
+                    Errors.Add("IOException while reading a section of the request: " + ioex.Message);
+                }
+            }
+            else
+            {
+                if (request.ContentType == null)
+                {
+                    Errors.Add("Part number 1 doesn't have a content type");
```

Comment:
This is not an error. Clients do not have to provide instance-template.

Reply:
The test might be entirely unnecessary. I don't think there are many http clients that would create requests without content-type. I added the test for completeness as we do test for content-type in each section when it is a multi part request.